### PR TITLE
Implement L1 solver for GLM

### LINKED ISSUE #9428: elastic_net implementation
Sometimes elastic_net gives a different result than the l1 and l1_cvxopt solvers:

```
import numpy as np
import statsmodels.api as sm
from sklearn.linear_model import LogisticRegression

# Load and prepare data
FACTOR = 1
X = np.array([[1, 4], [2, 1], [3, 3], [4, 2]] * FACTOR)
y = np.array([0, 0, 1, 1] * FACTOR)
X_c = sm.add_constant(X)
n = len(X)

ALPHA = .1
alpha_mask = np.ones(X_c.shape[1])
alpha_mask[0] = 0

# Sklearn with elastic net
model = LogisticRegression(
    penalty="elasticnet",
    solver="saga",
    max_iter=10000,
    C=1 / ALPHA / n,
    l1_ratio=1.0,
    warm_start=True,
    tol=1e-8,
)

model.fit(X, y)
print("sklearn saga:", model.intercept_, model.coef_)
coef = np.concatenate([model.intercept_, model.coef_[0]])
y_pred = model.predict_proba(X)[:, 1]
loss = -np.mean(y * np.log(y_pred) + (1 - y) * np.log(1 - y_pred))
print("Logistic regression loss:", loss + ALPHA * np.abs(model.coef_).sum())


# Statsmodels Logit with L1 (cvxopt)
model = sm.Logit(y, X_c)
results = model.fit_regularized(
    method="l1_cvxopt_cp", alpha=ALPHA * n * alpha_mask, disp=False, maxiter=100
)
print("Logit l1_cvxopt_cp:", results.params)
loss = -model.loglikeobs(results.params).mean()
print("Logit l1_cvxopt_cp loss:", loss + ALPHA * np.abs(results.params[1:]).sum())

# Statsmodels Logit with L1
model = sm.Logit(y, X_c)
results = model.fit_regularized(method="l1", alpha=ALPHA * n * alpha_mask, disp=False)
print("Logit l1:", results.params)
loss = -model.loglikeobs(results.params).mean()
print("Logit l1 loss:", loss + ALPHA * np.abs(results.params[1:]).sum())

# Statsmodels GLM with elastic net
model = sm.GLM(y, X_c, family=sm.families.Binomial())
results = model.fit_regularized(
    method="elastic_net", L1_wt=1.0, alpha=ALPHA * alpha_mask, maxiter=10000
)
print("GLM elastic_net (corrected):", results.params)
loss = -model.loglike(results.params) / n
print("GLM elastic_net loss:", loss + ALPHA * np.abs(results.params[1:]).sum())
```

```
sklearn saga: [-5.22609283] [[2.02421525 0.07529035]]
Logistic regression loss: 0.3804874930269413
Logit l1_cvxopt_cp: [-5.22611898  2.02422238  0.07529335]
Logit l1_cvxopt_cp loss: 0.38048749302397933
Logit l1: [-5.22610827  2.02422335  0.07528856]
Logit l1 loss: 0.3804874930259116
GLM elastic_net (corrected): [-5.09474961  2.03789985  0.        ]
GLM elastic_net loss: 0.3808717013601821
```

It is interesting because the objective is very similar but the mask of coefficients is not the same.

### PR DESCRIPTION
**Pull request by louisabraham**
- [x] closes #9428
- [x] tests added / passed.
- [x] code/documentation is well formatted.
- [x] properly formatted commit message. See
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message).

Adds `method="l1_slsqp"` to `GLM.fit_regularized`, reusing the `fit_l1_slsqp` solver from `statsmodels.base.l1_slsqp` (the same one used by the discrete models) through a new `_l1_slsqp` helper in `statsmodels/base/elastic_net.py`. In contrast to the coordinate descent algorithm used by `elastic_net`, this solves a smooth constrained reformulation of the L1 problem (interior point style); it only supports the lasso penalty (`L1_wt = 1`).

Tests compare the results against the discrete models' L1 `fit_regularized` (`Logit`, `Poisson`), against `elastic_net` with `L1_wt=1`, and against `OLS.fit_regularized` for the Gaussian family.

### CODE REVIEW

**Comment by josef-pkt** [comment]
don't blacken the module. I will not merge it.

**Comment by bashtage** [comment]
We should black the entire repo, but that is a bigger discussion.

**Comment by louisabraham** [comment]
> Any chance you could unblack the fine? Really hard to find genuine changes.

Just did it and repushed/

**Comment by louisabraham** [comment]
Hello! did you have time to look at this?

**Comment by louisabraham** [comment]
Thanks for the review! I rebased on main and pushed an update addressing the comments.

**Unit tests** (in `TestRegularized` in `test_glm.py`):
- `l1_slsqp` vs the corresponding discrete models (`Logit`, `Poisson` with `method="l1"`), using the existing `enet_binomial.csv` / `enet_poisson.csv` data — results agree to ~1e-5. One convention note: GLM's `fit_regularized` penalizes the average log-likelihood while the discrete models penalize the total, so `alpha_glm = alpha_discrete / nobs`; the tests encode this.
- `l1_slsqp` vs `elastic_net` with `L1_wt=1` (same objective, different algorithm) — close agreement on binomial and poisson.
- Gaussian family (a family with estimated scale) vs `OLS.fit_regularized(L1_wt=1)`.
- `refit=True`, the non-convergence warning, and the error paths.

**Reuse from the discrete models**: the implementation reuses `fit_l1_slsqp` from `statsmodels.base.l1_slsqp` — the same solver function the discrete models use — via a small `_l1_slsqp` helper in `statsmodels/base/elastic_net.py`, next to `_coord_descent` (the coordinate descent code extracted from `fit_elasticnet`). So the solver is fully shared and only thin glue is new (objective and score scaled by nobs, kwds handling). I didn't route through `DiscreteModel.fit_regularized` itself because it is tied to the discrete fit machinery (`extra_fit_funcs`, `cov_params_func_l1`, the L1 results classes). Since the helper lives in `base`, the other models that use `fit_elasticnet` (`PHReg`, `ConditionalLogit`/`ConditionalPoisson`) could get the same option with a one-line change — happy to do that here or in a follow-up.

**Results class**: `method="l1_slsqp"` returns the same `RegularizedResults`/`RegularizedResultsWrapper` as the elastic_net path (and the usual refit results with `refit=True`), so the two methods of `GLM.fit_regularized` behave consistently. I did not adopt the discrete `L1*Results` classes with their asymptotic inference.

**Naming**: the method is exposed under the full name `l1_slsqp` as requested (the `l1` shorthand is gone), and the docstrings/comments now state that it solves a smooth constrained reformulation of the L1 problem (interior point style), in contrast to the coordinate descent used by elastic_net.

Two bugs also got fixed along the way: `loglike_kwds`/`score_kwds` were not actually passed to `loglike`/`score` in the slsqp path (this is what the code-scanning "unused variable" warnings pointed at — `scale=1` was silently dropped), and a leftover `defaults` assignment was overwriting the method-specific defaults. The rebase also removes the old formatting noise, so the diff should be much easier to review now.
